### PR TITLE
fix: changing directory structure to idiomatic go directory format

### PR DESCRIPTION
**Pull request by tjgurwara99**
As the title suggests, I wrote this PR because there are some things within this repo that are not idiomatic to Go style of writing software. I changed the directory structure such that all the underscore characters `_` are removed from the package naming as well as changed some code around so that it is more readable and learners can use their own go code to experiment with these structures - main reason why I did a previous restructure #266 

Also fixed one of the LGTM analysis alerts which gets this repos alerts to zero.

### CODE REVIEW

**Comment by raklaptudirm** [comment]
Please re-request a review when this is done.

**Comment by tjgurwara99** [comment]
The action will continue to fail until the PR #338 is merged and I have resolved its conflicts.

**Review by raklaptudirm — DISMISSED** [review]

File: structure/binarytree/binarytree.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	fmt.Print("\n")
+	preorder(t.root)
+	fmt.Print("\n")
+	postorder(t.root)
+	fmt.Print("\n")
+	levelorder(t.root)
+	fmt.Print("\n")
+	fmt.Print(t.depth(), "\n")
+}
+*/
```
Comment (siriak) [inline structure/binarytree/binarytree.go:26]: Please remove that

**Review by siriak — APPROVED** [review]
LGTM, great job! Could you address my comment asap so that this PR is merged faster and other people can start working with the new structure?

File: structure/binarytree/binarytree.go (hunk 1 above)
Comment (tjgurwara99) [reply structure/binarytree/binarytree.go]: Done.

**Review by siriak — APPROVED** [review]
Great, thanks!

**Pull request merged**